pkg/api/customization/logging/list_handler.go

```diff
@@ -86,47 +95,48 @@ func convertToClusterResource(schema *types.Schema, clusterLogging v3.ClusterLog
 }
 
 func (h *ClusterLoggingHandler) setEmbeddedEndpoint(cl *v3.ClusterLogging) error {
-	if cl.Spec.EmbeddedConfig != nil {
-		esLabels := fmt.Sprintf("%s=%s", loggingconfig.LabelK8sApp, loggingconfig.EmbeddedESName)
-		espods, err := h.CoreV1.Pods(loggingconfig.LoggingNamespace).List(metav1.ListOptions{LabelSelector: esLabels})
-		if err != nil {
-			return err
-		}
-		esservice, err := h.CoreV1.Services(loggingconfig.LoggingNamespace).Get(loggingconfig.EmbeddedESName, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		if len(esservice.Spec.Ports) == 0 {
-			return fmt.Errorf("get service %s node port failed", loggingconfig.EmbeddedKibanaName)
-		}
-		var esPort int32
-		for _, v := range esservice.Spec.Ports {
-			if v.Name == "http" {
-				esPort = v.NodePort
-				break
-			}
+	espod, err := h.PodLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
```

Comment:
pod name is a random string as we use deployment. Please change it to use selector.